### ISSUE TITLE
front: "margin" component doesn't inform the user that it has changed

### ISSUE THREAD

**Issue by celine-sncf** [OpenRailAssociation/osrd#6079]
### What happened?

When inputing a margin in a train, then going back and chaging the rolling stock, the margin component doesn't turn to red to inform the user that it has automatically been reset
[marge disappear.webm](https://github.com/osrd-project/osrd/assets/115472753/f56e9dfc-c616-4d53-a5c6-fc5a0a3156fe)

### What did you expect to happen?

"margin" button should be highlighted in red like the "itinerary" button

### How can we reproduce it (as minimally and precisely as possible)?

1. go to study
2. create a train
3. add a margin
4. change the rolling stock
5. observe that the "itinerary" button is circled in red but not the "margin button"
6. go to "margins" to see your margins disappeared

### What operating system, browser and environment are you using?

- **Browser**: Chrome Version 119.0.6045.199
- **OS**: Windows 11
- **Env**: https://rec-osrd.reseau.sncf.fr/

### OSRD version (top right corner `Account` button > `Informations`)

staging b59698b

**Comment by SharglutDev** [comment]
Checked this issue with @thibautsailly, the tab should be highlighted in orange because it's only a warning.
Ideally, the tab should highlight only if there was a margin before reset, not all the time.

Same behavior for the simulation parameters tab.

**Comment by nicolaswurtz** [comment]
I agree with that, so it needs a bit of design, and it's not a bug! Moved to PM board.

**Comment by clarani** [comment]
The component Margin has been removed and replaced by the inputs/outputs table. We can close this issue
